extensions/rendering-hints/README.md

```diff
@@ -0,0 +1,63 @@
+# Rendering Hints Extension Specification
+
+- **Title: Rendering Hints**
+- **Identifier: rendering_hints**
+- **Field Name Prefix: render**
+- **Scope: Item**
+- **Extension [Maturity Classification](../README.md#extension-maturity): Proposal**
+
+One of the things many people want to do with data cataloged in STAC is to render it, often doing so directly
+from the assets (usually [Cloud Optimized GeoTIFF's](http://cogeo.org)) with dynamic web tile services. To 
+perform well these services benefit from a few 'hints' that can help the renderer perform more effectively. 
+
+This document explains the fields of the STAC Rendering Hints (`render`) Extension to a STAC Item. They are meant
+to be 'hints' - things that could be discovered by inspecting the data, but provided in STAC to help out tools
+that can take advantage of them. It is written with geospatial raster data in mind, but could apply to other data types.
+
+The field names defined herein should be added as fields in the Item Properties object. When specified on an Item, 
+the values are assumed to apply to all non-thumbnail Assets in that Item. 
+
+## Examples
+
+- TODO
+
+## Schema
+
+- TODO
+
+## Item Properties fields
+
+| Field Name       | Type                     | Description |
```

Comment:
(throwing this here because it seems like the most appropriate place to ask about included fields)

I don't know what the appropriate scope is for this extension, but I think the main hint I would want, if it's available, is a colormap from the tif. I'm a little bit confused about what I'd do with the min and max zoom hints.

Reply:
I've been interested in having that type of stuff in here, though I'm far from an expert on it. I think it can be in scope, and I was anticipating it'd grow to include similar things. But I'm more than happy to include it now. Just give me more details of what that actually looks like - what is the field, what's in it, how do you explain it. Or (preferred) feel free to add a 'commitable suggestion' (or whatever it is called) on the PR and I'll add it in.

Reply:
> I'm a little bit confused about what I'd do with the min and max zoom hints

Min and max zoom hints are helpful for on-demand rendering. The `max_zoom` is generally discoverable from the `gsd` separately.

That gets me thinking... what if a different way to describe min/max zoom is the `gsd` of the full-resolution data vs the `gsd` of the smallest overview? the main `gsd` is of course already stored per band. Storing the `min_overview_gsd` [of each band?] would allow the dynamic tiler to derive the min/max zoom levels, and would additionally support non-mercator rendering by being able to derive those zoom levels from the `gsd` values

Reply:
Additionally, for visualization it's sometimes important to know the min/max values (not zoom, actual values) for the data. Sometimes that can't be well determined from the data_type. For example, normalized_differences are usually between -1 and 1, but non of the data types really caters for that.

Additionally, I would like to use some of the fields from this extension, but don't always care about zoom levels. So having that required seems to make this less useful. But I'm not exactly sure what to require. Maybe just one of the properties? (Schema: `minProperties: 1`)

Reply:
min_overview_gsd does seem a bit cleaner. Is there an easy way to calculate that? Would want to give people some reference of how to get it. 

min/max values sounds good to add. I'm for it. What should the field name be? 

And `minProperties: 1` sounds good - I only knew the one use case, but if you have a use case that involves not using them then I'm happy to not make that one required.

Reply:
Responded in main thread.